sdk/confidentialledger/azure-confidentialledger/azure/confidentialledger/receipt/_receipt_verification.py

```diff
@@ -0,0 +1,269 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+"""Algorithm implementation for verifying Azure Confidential Ledger write
+transaction receipts."""
+
+from base64 import b64decode
+from hashlib import sha256
+from typing import Dict, List, Any, cast
+
+from cryptography.x509 import load_pem_x509_certificate, Certificate
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import ec, utils
+from cryptography.hazmat.primitives.serialization import Encoding, PublicFormat
+
+from azure.confidentialledger.receipt._models import (
+    LeafComponents,
+    ProofElement,
+    Receipt,
+)
+
+from azure.confidentialledger.receipt._utils import (
+    _convert_dict_to_camel_case,
+    _validate_receipt_content,
+)
+
+
+def verify_receipt(receipt: Dict[str, Any], service_cert: str) -> None:
```

Comment:
We're no longer supporting the `Receipt` model as input type here? I guess the receipts from the SDK will always be in JSON format?

Reply:
Correct. Johan pointed out that models should not be exposed unless there's some valid reason to do so and I decided to remove it (see [previous comment](https://github.com/Azure/azure-sdk-for-python/pull/27265#discussion_r1070099647)). Receipts will always be received in JSON format, so supporting JSON only should be enough for the moment.